File: programmers/new/level2/멀쩡한_사각형.java
class Solution {
	public long solution(int w,int h) {
        int gcd = getGCD(w,h);
        int sw = w / gcd;
        int sh = h / gcd;
        int dCount = sw+sh-1;
		return (long)w*h - (long)dCount * w / sw;
	}
    int getGCD(int a, int b){
        while(b > 0){
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
}
public class 멀쩡한_사각형 {
	public static void main(String[] args) {
		Solution sol = new Solution();
	}
}

File: programmers/js/new/level2/멀쩡한_사각형.js
function solution(w, h) {
    const gcd = getGCD(w, h);
    let count = (w + h) / gcd - 1;
    return w * h - count * gcd;
}

const getGCD = (a, b) => {
    if (a < b) {
        const temp = a;
        a = b;
        b = temp;
    }
    while (b > 0) {
        const temp = a % b;
        a = b;
        b = temp;
    }
    return a;
};
